### PR TITLE
VIVI-7773 change yt-dlp method used to extract info

### PR DESCRIPTION
**Pull request by keith-vivi**
This allows us to upgrade the version of `yt-dlp` we use

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -151,10 +151,13 @@ module.exports.processV2 = (output, origin) => {
...
 module.exports.processPlaylist = (output) => {
-  return output.toString().trim().split('\n').map((string) => {
-    const video = JSON.parse(string);
-    return `https://www.youtube.com/watch?v=${video.id}`;
-  });
+  let outputJSON = JSON.parse(output)
+  if (outputJSON["entries"]) {
+    entries = outputJSON["entries"]
+    return entries.map((video) => {
+      return `https://www.youtube.com/watch?v=${video.id}`;
```
Comment (tristanpenman) [inline index.js]: Can this be made an arrow function expression?

File: index.js [hunk 2]
```diff
@@ -151,10 +151,13 @@ module.exports.processV2 = (output, origin) => {
 };
 
 module.exports.processPlaylist = (output) => {
-  return output.toString().trim().split('\n').map((string) => {
-    const video = JSON.parse(string);
-    return `https://www.youtube.com/watch?v=${video.id}`;
-  });
+  let outputJSON = JSON.parse(output)
```
Comment (tristanpenman) [inline index.js]: const?

Comment (keith-vivi) [reply index.js]: Yep should be const. Fixed

File: index.js (hunk 1 above)
Comment (keith-vivi) [reply index.js]: Converted to arrow function expression. Kept the brackets around the param to match the pattern for other arrow function expressions in the same file.

**Comment by tristanpenman** [comment]
Were these changes meant to be pushed?

**Comment by keith-vivi** [comment]
> Were these changes meant to be pushed?

big facepalm yep

File: index.js [hunk 3]
```diff
@@ -151,10 +151,10 @@ module.exports.processV2 = (output, origin) => {
...
 module.exports.processPlaylist = (output) => {
-  return output.toString().trim().split('\n').map((string) => {
-    const video = JSON.parse(string);
-    return `https://www.youtube.com/watch?v=${video.id}`;
-  });
+  const outputJSON = JSON.parse(output)
+  if (outputJSON["entries"]) {
+    return outputJSON["entries"].map((video) => `https://www.youtube.com/watch?v=${video.id}`);
+  }
 };
```
Comment (tristanpenman) [inline index.js:159]: I think this should return an empty array if `outputJSON["entries"]` is not set.

Comment (keith-vivi) [reply index.js:159]: Yeah that would be more consistent with the old implementation. Fixed.

File: service.py [hunk 4]
```diff
@@ -5,20 +5,17 @@
 from sys import stderr
 from urllib.parse import parse_qs, urlparse
 from yt_dlp import YoutubeDL
+import json
 
 class Handler(BaseHTTPRequestHandler):
     error_message_format = '%(explain)s'
 
     def __init__(self, *args, **kwargs):
```
Comment (tristanpenman) [inline service.py]: Is this now redundant, because it doesn't do anything other than call the super constructor?

Comment (keith-vivi) [reply service.py]: Yep. Removed the constructor and `error_message_format`. I can't find a point in the history where `error_message_format` was ever used.

**Review by tristanpenman — APPROVED** [review]

**Pull request merged**